api/tests/unit/integrations/lead_tracking/hubspot/_hubspot_responses.py

```diff
@@ -0,0 +1,106 @@
+from datetime import datetime
+
+from dateutil.tz import tzlocal
+
+
+class DummyHubspotResponse:
+    """
+    Dummy class to replicate the to_dict() method of the Hubspot response classes.
+    """
+
+    def __init__(self, data: dict) -> None:
+        self.data = data
+
+    def to_dict(self) -> dict:
+        return self.data
+
+
+def generate_get_company_by_domain_response(
+    name: str, domain: str
+) -> DummyHubspotResponse:
+    """
+    Generate a sample response given by the Hubspot API when searching for a company by domain.
+
+    This response was retrieved from the API directly, and then modified to allow us to set the
+    certain properties dynamically.
+    """
+
+    return DummyHubspotResponse(
+        data={
+            "paging": None,
+            "results": [
+                {
+                    "archived": False,
+                    "archived_at": None,
+                    "created_at": datetime(
+                        2024, 1, 25, 21, 48, 28, 655000, tzinfo=tzlocal()
+                    ),
+                    "id": "9765318341",
+                    "properties": {
+                        "createdate": "2024-01-25T21:48:28.655Z",
+                        "domain": domain,
+                        "hs_lastmodifieddate": "2024-04-23T15:54:13.336Z",
+                        "hs_object_id": "9765318341",
+                        "name": name,
+                    },
+                    "properties_with_history": None,
+                    "updated_at": datetime(
+                        2024, 4, 23, 15, 54, 13, 336000, tzinfo=tzlocal()
+                    ),
+                }
+            ],
+            "total": 1,
+        }
+    )
+
+
+def generate_get_company_by_domain_response_no_results() -> DummyHubspotResponse:
+    """
+    Generate a sample response given by the Hubspot API when searching for a company by domain
+    but no results are returned.
+
+    This response was retrieved from the API directly and hard coded here for simplicity.
+    """
+
+    return DummyHubspotResponse(
+        data={
+            "paging": None,
+            "results": [],
+            "total": 0,
+        }
+    )
+
+
+def generate_create_company_response(
+    name: str, domain: str = None, organisation_id: int = None
```

Comment:
You're missing the typing for the `None` values here, but in addition to that I'm not sure if organisation should be set null since the default case as noted about is actually the value `-1`, right?

Reply:
Yep, good spot. Fixed.